catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/QueryApplication.java

```diff
@@ -126,7 +136,30 @@ private CqlQueryResponse executeCqlQuery(CqlRequest cqlRequest)
                 filterBuilder);
 
         Stopwatch stopwatch = Stopwatch.createStarted();
-        QueryResponse response = catalogFramework.query(request);
+        QueryResponse response = null;
+
+        if (configuration.isDisableLocalCatalog()) {
+            Set<String> sourceIds;
+            if (CollectionUtils.isNotEmpty(request.getSourceIds())) {
+                sourceIds = new HashSet<>(request.getSourceIds());
+            } else {
+                sourceIds = new HashSet<>(catalogFramework.getSourceIds());
+            }
+            sourceIds.remove(catalogFramework.getId());
+            request = new QueryRequestImpl(request.getQuery(),
+                    false,
+                    sourceIds,
+                    request.getProperties());
+
+            if (CollectionUtils.isNotEmpty(sourceIds)) {
+                response = catalogFramework.query(request);
```

Comment:
It's a nit, but since we call `response = catalogFramework.query(request);` regardless, this added logic could just set the `QueryResponse response` conditionally.